### PR TITLE
[Merged by Bors] - ci: determine Zulip maintainer merge topic from PR labels

### PR DESCRIPTION
**Pull request by jcommelin**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by adomani** [comment]
As a general comment, how about extracting the code to a script and then running the script as `./scripts/dispatch_label.sh` in each workflow?

**Comment by jcommelin** [comment]
maintainer merge

**Comment by jcommelin** [comment]
> As a general comment, how about extracting the code to a script and then running the script as `./scripts/dispatch_label.sh` in each workflow?

Hmm, I think the scripts are short enough that it isn't really worth it.

File: scripts/get_tlabel.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+tlabels="$(gh api --jq '.labels.[].name' "${PR}" | grep -- '^t-' || printf 'generic')"
+# print to error, since the stdout is captured into `GITHUB_OUTPUT
+>&2 printf 't-labels:\n---\n%s\n---\n' "${tlabels}"
+# if there isn't exactly 1 `t-xxx` label, use `generic`
+if [[ "$(wc -l <<<"${tlabels}")" -ne 1 ]]; then
+  tlabels="generic"
+fi
+streamName="${tlabels} maintainer merges"
+>&2 printf $'Post to topic: \'%s\'"\n' "${streamName}"
+echo "topic=${streamName}"
```
Comment (jcommelin) [inline scripts/get_tlabel.sh:16]: In hindsight, I think I prefer all the topics starting with `maintainer merge` for visual organization/alignment purposes. But that's a minor concern.
```suggestion
topicName="$maintainer merge: {tlabels}"
>&2 printf $'Post to topic: \'%s\'"\n' "${topicName}"
echo "topic=${topicName}"
```

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**